pkg/e2e/framework/nodes.go

```diff
@@ -15,6 +17,22 @@ func AddNodeCondition(c client.Client, node *corev1.Node, cond corev1.NodeCondit
 	return c.Status().Patch(context.Background(), nodeCopy, client.MergeFrom(node))
 }
 
+// GetNodeForMachine retrieves the node backing the given Machine.
+func GetNodeForMachine(c client.Client, m *mapiv1beta1.Machine) (*corev1.Node, error) {
```

Comment:
There're helpers for this in infra/utils `getNodeFromMachine` already
We'll want to consolidate the reusable library in the upcoming follow ups.

Reply:
Yeah for sure. That's one of the frustrating things I ran into. There's also a bunch of helpers there, but almost none of them are exported for some reason.